api/src/main/java/com/capitalone/dashboard/webhook/github/GitHubCommitV3.java

```diff
@@ -0,0 +1,318 @@
+package com.capitalone.dashboard.webhook.github;
+
+import com.capitalone.dashboard.ApiSettings;
+import com.capitalone.dashboard.client.RestClient;
+import com.capitalone.dashboard.custom.GitHubParsed;
+import com.capitalone.dashboard.misc.HygieiaException;
+import com.capitalone.dashboard.model.CollectorItem;
+import com.capitalone.dashboard.model.Commit;
+import com.capitalone.dashboard.model.CommitType;
+import com.capitalone.dashboard.model.GitRequest;
+import com.capitalone.dashboard.model.webhook.github.GitHubRepo;
+import com.capitalone.dashboard.repository.CommitRepository;
+import com.capitalone.dashboard.repository.GitHubRepoRepository;
+import com.capitalone.dashboard.repository.GitRequestRepository;
+import com.capitalone.dashboard.service.CollectorService;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.joda.time.DateTime;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.ParseException;
+import org.springframework.data.querydsl.QueryDslPredicateExecutor;
+import org.springframework.http.ResponseEntity;
+
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class GitHubCommitV3 extends GitHubV3 {
+    private static final Log LOG = LogFactory.getLog(GitHubCommitV3.class);
+
+    private final CommitRepository commitRepository;
+    private final GitRequestRepository gitRequestRepository;
+    private final GitHubRepoRepository gitHubRepoRepository;
+
+    public GitHubCommitV3(CollectorService collectorService,
+                          RestClient restClient,
+                          CommitRepository commitRepository,
+                          GitRequestRepository gitRequestRepository,
+                          GitHubRepoRepository gitHubRepoRepository,
+                          ApiSettings apiSettings) {
+        super(collectorService, restClient, apiSettings);
+
+        this.commitRepository = commitRepository;
+        this.gitRequestRepository = gitRequestRepository;
+        this.gitHubRepoRepository = gitHubRepoRepository;
+    }
+
+    @Override
+    public QueryDslPredicateExecutor<GitHubRepo> getGitHubRepoRepository() { return this.gitHubRepoRepository; }
+
+    @Override
+    public String process(JSONObject jsonObject) throws MalformedURLException, HygieiaException, ParseException {
+        String result = "Commits Processed Successfully";
+
+        String branch = null;
+        String repoUrl = null;
+
+        Object commitsObj = jsonObject.get("commits");
+        if ((commitsObj == null) || !(commitsObj instanceof List)) {
```

Comment:
just checking instanceof should be sufficient here

Reply:
Have taken care of this